ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -219,16 +219,13 @@ void addAll(Collection<LogAppender> newSenders) {
       Preconditions.assertTrue(changed);
     }
 
-    boolean removeAll(Collection<LogAppender> c) {
+    synchronized boolean removeAll(Collection<LogAppender> c) {
```

Comment:
We don't need `synchronized` for both `addAll` and `removeAll` since it uses a `CopyOnWriteArrayList`.